src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OverrideCompletionProviderTests.cs

```diff
@@ -2700,6 +2700,148 @@ public override bool Equals(object obj)
             }
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task CommitRequiredKeywordAdded()
+        {
+            var markupBeforeCommit = """
+                class Base
+                {
+                    public virtual required int Prop { get; }
+                }
+
+                class Derived : Base
+                {
+                    override $$
+                }
+                """;
+
+            var expectedCodeAfterCommit = """
+                class Base
+                {
+                    public virtual required int Prop { get; }
+                }
+
+                class Derived : Base
+                {
+                    public override required int Prop
+                    {
+                        get
+                        {
+                            return base.Prop;$$
+                        }
+                    }
+                }
+                """;
+            await VerifyCustomCommitProviderAsync(markupBeforeCommit, "Prop", expectedCodeAfterCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task CommitRequiredKeywordPreserved()
+        {
+            var markupBeforeCommit = """
+                class Base
+                {
+                    public virtual required int Prop { get; }
+                }
+
+                class Derived : Base
+                {
+                    required override $$
+                }
+                """;
+
+            var expectedCodeAfterCommit = """
+                class Base
+                {
+                    public virtual required int Prop { get; }
+                }
+
+                class Derived : Base
+                {
+                    public override required int Prop
+                    {
+                        get
+                        {
+                            return base.Prop;$$
+                        }
+                    }
+                }
+                """;
+            await VerifyCustomCommitProviderAsync(markupBeforeCommit, "Prop", expectedCodeAfterCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task CommitRequiredKeywordRemovedForMethods()
```

Comment:
Is this behavior consistent with other modifiers like this? It almost feels like override completion should instead be filtering to required members in that case, i.e. I wrote "required" why is it being ignored and erased?

Reply:
I don't necessarily think so. Maybe we could make it filter out things that _can't_ be required, but I might want the override of a non-required property to be required by the override.